dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/VersionData.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.cloudbomdashboard;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+
+import org.apache.commons.io.FileUtils;
+
+import org.eclipse.aether.artifact.Artifact;
+
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+import java.net.URL;
+import java.net.URLConnection;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.time.LocalDateTime;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.Scanner;
+import java.util.Date;
+
+/**
+ * Container class for the conversion of a dashboard into a template
+ */
+public class VersionData {
+    /* There can only be one version of our 'All Versions' page */
+    public static final String ALL_VERSIONS_NAME = "all-versions";
+    private static final VersionData ALL_VERSIONS_DATA = new VersionData();
+
+    /* Helps to improve performance, so we don't have to repeatedly look up
+        remote resources. */
+    private static final Map<String, String> pomToDepsVersion = new HashMap<>();
+    private static final Map<Artifact, String> artifactToTime = new HashMap<>();
+    private static final Map<Artifact, String> artifactToLatestVersion = new HashMap<>();
+
+    private final List<String> versions = new ArrayList<>();
+
+    /* Everything associated with the template for this VersionData */
+    private final Set<String> artifacts = new TreeSet<>();
```

Comment:
Unfortunately this seems to be the most straightforward way to do this from a code perspective, otherwise I believe the only way around it may be to do some aggressive formatting in the Freemarker file. Unfortunately for each artifact, we need to store 10 pieces of data - java-cloud-bom version, artifact ID, artifact version, last release date, version, last released version, and java-shared-dependencies, plus links to each of the middle three columns